### ISSUE TITLE
Serve: There is no way to build a deployment graph across applications

### ISSUE THREAD

**Issue by tchordia** [ray-project/ray#40242]
### Description

1. There is no recommended, race condition free api to build a deployment graph across application boundaries.

The problem:

I have 1 router deployment, and N model deployments. I want to be able to update the models in place, without affecting the other models.

The only way to do this is to create an application for each deployment, and then loop pass the name of the application into the router. The router then tries to get the app of the same name, and loop until it can. 

A preferrable API in my opinion:

```
model_1_handle = serve.app(Model1.bind(...))
model_2_handle = serve.app(Model2.bind(...))
router_handle = Router.bind(model_1_handle, model_2_handle)
serve.run(router_handle)
```

Now each model will be treated as a separate application. Importantly, all applications are deployed together, and atomically [so no weird behavior and no race conditions].

**Comment by abrarsheikh** [comment]
i dont think I understand the problem you are trying to solve. Could you rephrase your question to include
1. What is that your are you trying to achieve or solve?
2. Why is the current serve api restrictive or does not solve your problem? Include some code that demonstrates the problem(race condition free api to build)

**Comment by zcin** [comment]
Closing, please re-open if necessary.
